### PR TITLE
[SIP-5] Repair and refactor Horizon Chart

### PR DESCRIPTION
**Pull request by kristw**
- Rewrite into React Component.
- Fix visual issues
- Enable `renderTrigger` on compatible props

Before

image

After

image

@williaster @conglei

### CODE REVIEW

File: superset/assets/src/explore/controls.jsx [hunk 1]
```diff
@@ -383,14 +383,15 @@ export const controls = {
...
+    renderTrigger: true,
+    label: t('Value Domain'),
     choices: [
       ['series', 'series'],
       ['overall', 'overall'],
       ['change', 'change'],
     ],
     default: 'series',
-    description: t('Defines how the color are attributed.'),
+    description: t('series: Treat each series independently; overall: All series use the same scale; change: Show changes compared to the first data point in each series'),
```
Comment (williaster) [inline superset/assets/src/explore/controls.jsx]: 💯

File: superset/assets/src/visualizations/HorizonRow.jsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  offsetX: PropTypes.number,
+  title: PropTypes.string,
+  yDomain: PropTypes.arrayOf(PropTypes.number),
+};
+
+const defaultProps = {
+  className: '',
+  width: 800,
+  height: 20,
+  bands: DEFAULT_COLORS.length >> 1,
```
Comment (williaster) [inline superset/assets/src/visualizations/HorizonRow.jsx]: any way to write this without bitwise operators for readability? like `Math.floor(DEFAULT_COLORS.length / 2)` or something?

**Review by williaster — APPROVED** [review]
This is much better! Thanks for refactoring 👍 Had one comment but otherwise lgtm

File: superset/assets/src/visualizations/HorizonRow.jsx (hunk 2 above)
Comment (kristw) [reply superset/assets/src/visualizations/HorizonRow.jsx]: Addressed.

**Comment by kristw** [comment]
Ready to merge

**Review by williaster — APPROVED** [review]

**Pull request merged**